lib/Parser.php

```diff
@@ -236,16 +236,12 @@ public function getDocument() : Document
         return $this->document;
     }
 
-    public function includeFileAllowed(string $path) : bool
+    private function includeFileAllowed(string $path) : bool
     {
         if (! $this->includeAllowed) {
             return false;
         }
 
-        if (! @is_readable($path)) {
-            return false;
-        }
```

Comment:
Moved this so that it throws an exception. If an include is invalid, I think we should fail first. THEN we should check if the include is allowed.

Reply:
Disagree with this change, because the method is called `includeFileAllowed`, which should not bother about whether it is called before or after other bits of code: it just responds ro the simple question on whether the path can be loaded.